lib/mixins/repeatable.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var request = require('../request')
+  ;
+
+var Repeatable = {
+  repeat: {
+    value: function (id, payload) {
+      if (!id) {
+        throw {
+          name: 'ArgumentError',
+          message: id + ' is not a valid id'
+        };
+      }
+
+      if (!payload) {
+        throw {
+          name: 'ArgumentError',
+          message: payload + ' is not a valid payload'
+        };
+      }
+
+      return request({
+
+        method: 'post',
+        protocol: this.protocol && this.protocol + ':',
+        host: this.host,
+        port: this.port,
+        path: [this.path, id, 're_run'].join('/'),
+        version: this.api_version,
+        withCredentials: false,
+        headers: this.token ? {
+          'content-type': 'application/json',
+          'x-auth-token': this.token.token
+        } : {
+          'content-type': 'application/json'
+        },
+
+      }, payload).then(function (res) {
+
+        if (res.statusCode !== 201) {
```

Comment:
I don't remember where exactly, but I've already posted a link on a blog post regarding JS Error inheritance. The bottom line is that there's so much things that can go wrong with that, you should not rely on instanceOf and instead just check the name as it is the only thing that stays the same across multiple browsers and OOP implementations.